addons/pos_restaurant/static/src/js/Chrome.js

```diff
@@ -76,7 +76,8 @@ odoo.define('pos_restaurant.chrome', function (require) {
                 this.showScreen('FloorScreen', { floor: table ? table.floor : null });
             }
             _shouldResetIdleTimer() {
-                return super._shouldResetIdleTimer() && this.env.pos.config.iface_floorplan && this.mainScreen.name !== 'FloorScreen';
+                const stayPaymentScreen = this.mainScreen.name === 'PaymentScreen' && this.env.pos.get_order().paymentlines.length > 0;
+                return super._shouldResetIdleTimer() && this.env.pos.config.iface_floorplan && !stayPaymentScreen && this.mainScreen.name !== 'FloorScreen';
             }
             __showScreen() {
```

Comment:
Instead of putting `this.env.pos.config.iface_floorplan`, can you consider removing this override of `__showScreen`? You would notice that `_setActivityListeners` is the one that starts the "auto-FloorScreen" mechanism and it's only called when iface_floorplan is true [*].

I'm suggesting to remove this override because it is bypassing the condition before the IDLE_TIME_SETTER is called. Besides, "showing a new screen" is done normally by "clicking", therefore, the call to `_setIdleTimer` is a redundant call.

[*] Important sidenote: Perhaps you should make that condition more robust, such that it's only true if there are also floors.

Reply:
I was also thinking about removing the override of the function `__showScreen` in master.  Also, making the condition more robust is a good point. 

Do you want me to create another task in master for that or we can do it on the fly?

Reply:
It's going to be part of this task because it should be forward-ported. There is a conflict in the forward port to saas-15.4. Perhaps you can start the change there.